src/rentACar/Application/Features/Brands/Commands/UpdateBrand/UpdateBrandCommand.cs

```diff
@@ -36,10 +36,13 @@ public class UpdateBrandCommandHandler : IRequestHandler<UpdateBrandCommand, Upd
 
         public async Task<UpdatedBrandDto> Handle(UpdateBrandCommand request, CancellationToken cancellationToken)
         {
-            await _brandBusinessRules.BrandIdShouldExistWhenSelected(request.Id);
-            Brand mappedBrand = _mapper.Map<Brand>(request);
-            Brand updatedBrand = await _brandRepository.UpdateAsync(mappedBrand);
-            UpdatedBrandDto updatedBrandDto = _mapper.Map<UpdatedBrandDto>(updatedBrand);
+            var brand = await _brandRepository.GetAsync(x => x.Id == request.Id);
+            _brandBusinessRules.BrandIdShouldExistWhenSelected(brand);
+            await _brandBusinessRules.BrandNameCanNotBeDuplicatedWhenUpdated(request.Id, request.Name);
```

Comment:
You can use the query instance you already have to check for duplicates. Pass a parameter with the  `previousName` named. And compare the new value and the previous name with lower cases. If they are the same value, you can return the method.